src/main/java/io/quarkus/gizmo2/impl/Neg.java

```diff
@@ -26,12 +32,12 @@ public boolean bound() {
     }
 
     public void writeCode(final CodeBuilder cb, final BlockCreatorImpl block) {
-        switch (typeKind()) {
+        switch (typeKind().asLoadable()) {
             case INT -> cb.ineg();
             case LONG -> cb.lneg();
             case FLOAT -> cb.fneg();
             case DOUBLE -> cb.dneg();
-            default -> throw new IllegalStateException();
+            default -> throw impossibleSwitchCase(a);
```

Comment:
Normally you'd put in the case that was impossible here, i.e. `typeKind().asLoadable()`.

Reply:
Since `typeKind()` boils down to `a.typeKind()`, I figured reporting the whole `a` would be more informative here.

Reply:
Changed it to `typeKind().asLoadable()` in the end. Done.